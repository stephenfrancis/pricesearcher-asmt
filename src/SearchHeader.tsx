
import * as React from 'react';
import TextField from '@material-ui/core/TextField';


interface Props {
  setFullTextSearch: (full_text_search: string) => void;
}

interface State {
}

// Top level app component
export default class SearchHeader extends React.Component<Props, State> {

  constructor(props) {
    super(props);
    this.state = {};
    this.onChange = this.onChange.bind(this);
  }


  private onChange(event) {
    this.props.setFullTextSearch(event.target.value || null);
  }


  public render() {
    return (
      <div>
        <TextField
          placeholder="Search for..."
          onChange={this.onChange}
          type="text"
        />
      </div>
    );
  }

}
