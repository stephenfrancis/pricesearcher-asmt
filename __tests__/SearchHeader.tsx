
import * as React from "react";
import * as Renderer from "react-test-renderer";
import * as ReactTestUtils from 'react-dom/test-utils';
import { GridListTile, } from '@material-ui/core';

import SearchHeader from "../src/SearchHeader";


test("appearance", () => {

  const setFullTextSearch = jest.fn();

  const component = Renderer.create(
    <SearchHeader setFullTextSearch={setFullTextSearch} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot(); // modal closed

  const inputs = component.root.findAllByType("input");
  expect(inputs.length).toBe(1); // only 1 tiie component
  expect(setFullTextSearch.mock.calls.length).toBe(0); // not called yet

  // ReactTestUtils.Simulate.change(inputs[0], { target: { value: "some text" } } as any);

  // expect(setFullTextSearch.mock.calls.length).toBe(1);

  // tree.children[0].props.onClick();
  // ReactTestUtils.Simulate.click(tile[0].instance);

  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot(); // modal open
});
