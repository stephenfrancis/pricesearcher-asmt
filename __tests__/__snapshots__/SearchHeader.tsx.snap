// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appearance 1`] = `
<div>
  <div
    className="MuiFormControl-root-1"
  >
    <div
      className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-13"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search for..."
        type="text"
      />
    </div>
  </div>
</div>
`;
